### PR TITLE
Add cursor on nodes from Panel

### PR DESCRIPTION
**Pull request by guimatheus92**
### Description

Adding cursor on each node from Panel (tree), to improve UX when hovering each entity.

### How Has This Been Tested?

Open the panel and hover on each entity.

### Checklist

* [X] I have tested and built the changes locally and they work as expected
* [X] I have added relevant documentation or updated existing documentation
* [X] My changes generate no new warnings

### Screenshots (if applicable)

image

### Additional Context

@shmuelzon can you please check and review if code if closer to what we wanted to achieve please?
I think a better usability of this feature would be to apply only to entities, but I wasn't able to ignore the group itselft. Can you help please? Check image below:

image

### CODE REVIEW

**Comment by shmuelzon** [comment]
Hey, yes, this is great! Thanks!

I was able to get your hover/highlight thing to be applied only on entities so I'll add that later on. With this new feature, I also thing we can replace the double-click requirement with a single click since the entity is already highlighted and the curser changes and everything so I think it would be more intuitive and I'll add that to the pull request too.

I do have a couple of questions, though, so I'll add them to the review.

File: src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java [hunk 1]
```diff
@@ -193,6 +198,45 @@ protected void setExpandedState(TreePath path, boolean state) {
...
+                    label.setPreferredSize(new Dimension(label.getPreferredSize().width + 20, label.getPreferredSize().height));
+                } else {
+                    /* Reset to original font for non-leaf nodes (groups) */
+                    label.setFont(originalFont);
+                }
+
+                return label;
+            }
+        };
+
```
Comment (shmuelzon) [inline src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java:239]: Does this actually do anything? I haven't seen any case where the font becomes bold and, in your screenshots, the font looks the same for both nodes and leaves whether they are selected or not.

File: src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java [hunk 2]
```diff
@@ -203,13 +247,31 @@ public void mouseClicked(java.awt.event.MouseEvent event) {
                     return;
 
                 DefaultMutableTreeNode node = (DefaultMutableTreeNode)selectedPath.getLastPathComponent();
-                if (!node.isLeaf())
+                if (!node.isLeaf() || !(node.getUserObject() instanceof EntityNode)) {
```
Comment (shmuelzon) [inline src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java:250]: Why was this needed? Did you actually reach a case where a leaf wasn't an `EntityNode`?

File: src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java (hunk 1 above)
Comment (guimatheus92) [reply src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java:239]: Yes! Do set the current hightlight I had to all of this, otherwise the text would get cut at the end because of highlight. I don't know if you could improve this though.

File: src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java (hunk 2 above)
Comment (guimatheus92) [reply src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java:250]: I added to avoid any problem in the cursor, I didn't actually reach a case where a leaf wasn't an `EntityNode`, but I was being prepared if something happened. What do you think?

File: src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java (hunk 1 above)
Comment (shmuelzon) [reply src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java:239]: Not sure I see the difference, maybe an OS-dependent kind of thing? Can you attach a screenshot or something with the difference, please?

Comment (guimatheus92) [reply src/com/shmuelzon/HomeAssistantFloorPlan/Panel.java:239]: I just removed that whole code block and is still working, but I remember I struggled a bit on this. I don't know I got confused. I will push without this part then. It's working:

image
